### PR TITLE
Dshekera/tdi 42581 t soap doesnt support soap actions for soap 1 2

### PR DESCRIPTION
**Pull request by dmytroshekera**
https://jira.talendforge.org/browse/TDI-42581

**What is the new behavior?**

**Please check if the PR fulfills these requirements**

- [ ] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: main/plugins/org.talend.designer.components.libs/libs_src/soap/src/main/java/org/talend/soap/SOAPUtil.java [hunk 1]
```diff
@@ -124,7 +124,14 @@ public void invokeSOAP(String version, String destination, String soapAction, St
         }
         SOAPMessage message = messageFactory.createMessage();
         MimeHeaders mimeHeaders = message.getMimeHeaders();
-        mimeHeaders.setHeader("SOAPAction", soapAction);
+
+        // https://jira.talendforge.org/browse/TDI-42581 skip add SOAPAction directly to header v1.2
+        if (SOAP12.equals(version)) {
+            mimeHeaders.addHeader("Content-Type", "application/soap+_xml; charset=utf-8; action=\"" + soapAction+"\"");
```
Comment (slushatel) [inline main/plugins/org.talend.designer.components.libs/libs_src/soap/src/main/java/org/talend/soap/SOAPUtil.java]: I would use only addHeader or only setHeader. Not both.

File: main/plugins/org.talend.designer.components.libs/libs_src/soap/src/main/java/org/talend/soap/SOAPUtil.java [hunk 2]
```diff
@@ -263,7 +270,12 @@ public String extractContentAsDocument(String version, String destination, Strin
     	}
     	SOAPMessage message = messageFactory.createMessage();
     	MimeHeaders mimeHeaders = message.getMimeHeaders();
-    	mimeHeaders.setHeader("SOAPAction", soapAction);
+        if (SOAP12.equals(version)) {
```
Comment (slushatel) [inline main/plugins/org.talend.designer.components.libs/libs_src/soap/src/main/java/org/talend/soap/SOAPUtil.java]: duplicated code can be moved to the separate method

File: main/plugins/org.talend.designer.components.libs/libs_src/soap/src/main/java/org/talend/soap/SOAPUtil.java (hunk 1 above)
Comment (dmytroshekera) [reply main/plugins/org.talend.designer.components.libs/libs_src/soap/src/main/java/org/talend/soap/SOAPUtil.java]: change to setHeader

File: main/plugins/org.talend.designer.components.libs/libs_src/soap/src/main/java/org/talend/soap/SOAPUtil.java (hunk 2 above)
Comment (dmytroshekera) [reply main/plugins/org.talend.designer.components.libs/libs_src/soap/src/main/java/org/talend/soap/SOAPUtil.java]: extract method

**Pull request merged**